nlp-common/src/main/java/io/cdap/google/common/NLPMethodExecutor.java

```diff
@@ -62,20 +76,71 @@ public String execute(String text) {
     }
   }
 
+  public StructuredRecord executeAndReturnStructuredRecord(String text) {
+    return getRecordFromJson(execute(text));
+  }
+
   @Override
   public void close() {
     if (language != null) {
       language.close();
     }
   }
 
+  protected List<String> jsonArrayToList(JsonArray jsonArray) {
+    List<String> result = new ArrayList<>();
+    for (int i = 0; i < jsonArray.size(); i++) {
+      result.add(jsonArray.get(i).getAsJsonObject().toString());
+    }
+    return result;
+  }
+
+  protected abstract StructuredRecord getRecordFromJson(String json);
   protected abstract MessageOrBuilder executeRequest(LanguageServiceClient language, Document document);
 
+  protected static List<Map<String, Object>> flattenJsonObjects(JsonArray jsonArray) {
+    List<Map<String, Object>> results = new ArrayList<>();
+    for (int i = 0; i < jsonArray.size(); i++) {
+      results.add(flattenJsonObject(jsonArray.get(i).getAsJsonObject()));
+    }
+    return results;
+  }
+
+  protected static Map<String, Object> flattenJsonObject(JsonObject jsonObject) {
```

Comment:
this is not doing the right thing, a record field must be a StructuredRecord, not a `Map<String, Object>`.

Instead of trying to have a generic flattener, I think it's better to just explicitly build the record objects from the JsonObject in each individual transform

Reply:
Very nice catch. I missed this for some reason.